### PR TITLE
XGBoost Operator: Refactor manifests structure

### PR DESCRIPTION
**Pull request by yanniszark**
Umbrella-Issue: https://github.com/kubeflow/manifests/issues/1769

#### Current manifests structure

```
manifests
└── xgboost-operator
    ├── base
    └── overlays
        └── application

```

#### Explanation

- The `application` overlays contain Application CRs, which we want to move away from (https://github.com/kubeflow/manifests/issues/1715#issuecomment-779402779)

#### Recommended end state

```
manifests
├── base
└── overlays
    └── kubeflow

```

In the future, we'd like `base` to import the kubebuilder-generated manifests.

### CODE REVIEW

**Comment by yanniszark** [comment]
cc @kubeflow/wg-training-leads

File: README.md [hunk 1]
```diff
@@ -19,15 +19,9 @@ This repository contains the specification and implementation of `XGBoostJob` cu
...
+cd manifests
 kubectl create namespace kubeflow
-kustomize build base | kubectl apply -f -
-```
-
-Note that since Kubernetes v1.14, `kustomize` became a subcommand in `kubectl` so you can also run the following command instead:
-
-```bash
-kubectl kustomize base | kubectl apply -f -
+kustomize build overlays/kubeflow | kubectl apply -f -
```
Comment (terrytangyuan) [inline README.md:24]: Can we stick with `kubectl kustomize`?

Comment (yanniszark) [reply README.md]: @terrytangyuan sure!

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**